core/integration/services_test.go

```diff
@@ -1206,6 +1206,65 @@ func TestServiceStartStop(t *testing.T) {
 	require.Empty(t, out)
 }
 
+// TestServiceStartStopKill tests that we send SIGTERM by default, instead of SIGKILL.
+// Additionally, we check that we can attempt to SIGKILL a process that is not
+// responding to SIGTERM.
+func TestServiceStartStopKill(t *testing.T) {
+	t.Parallel()
+
+	c, ctx := connect(t)
+
+	httpSrv, httpURL := signalService(ctx, t, c)
+
+	fetch := func() (string, error) {
+		return c.Container().
+			From(alpineImage).
+			WithEnvVariable("BUST", identity.NewID()).
+			WithExec([]string{"wget", "-O-", httpURL + "/signals.txt"}).
+			Stdout(ctx)
+	}
+
+	out, err := fetch()
+	require.Error(t, err)
+	require.Empty(t, out)
+
+	_, err = httpSrv.Start(ctx)
+	require.NoError(t, err)
+
+	out, err = fetch()
+	require.NoError(t, err)
+	require.Empty(t, out)
+
+	eg := errgroup.Group{}
+	eg.Go(func() error {
+		// attempt to SIGTERM the service
+		// this won't stop the service though, since the process ignores this one - we'll
+		// end up blocking here until the SIGKILL applies
+		_, err = httpSrv.Stop(ctx)
```

Comment:
We don't really have any tests at the moment for services automatically closing.

https://github.com/dagger/dagger/blob/746339f5f89397dbf31247994605a6c2df493248/core/integration/services_test.go#L1172

@vito I had an idea where we could potentially spin up a dagger instance and watch the logs to see if the service exits? Not 100% sure how to approach this, but I think we should probably try and do it now, so as to not accidentally break that behavior in this patch.